File: src/routes/examples/norns/notes/+page.server.c
import { fail, redirect } from '@sveltejs/kit'
import { listNotes, createNote } from './db'

export load = ->
	notes: listNotes()

export actions =
	create: ({ request }) ->
		data = await request.formData()
		title = data.get('title')?.toString().trim()
		body = data.get('body')?.toString() ? ''
		return fail 400, { error: 'title is required', body } unless title
		id = createNote title, body
		throw redirect 303, "/examples/norns/notes/#{id}"

File: src/routes/examples/svelte/notes/+page.server.ts
import { fail, redirect } from '@sveltejs/kit';
import type { Actions, PageServerLoad } from './$types';
import { listNotes, createNote } from './db';

export const load: PageServerLoad = () => {
	return { notes: listNotes() };
};

export const actions: Actions = {
	create: async ({ request }) => {
		const data = await request.formData();
		const title = data.get('title')?.toString().trim();
		const body = data.get('body')?.toString() ?? '';
		if (!title) return fail(400, { error: 'title is required', body });
		const id = createNote(title, body);
		throw redirect(303, `/examples/svelte/notes/${id}`);
	}
};
